README.md

```diff
@@ -0,0 +1,165 @@
+plays, either in an OCP 4.x installation role or tasks. Note that because
+this role leaves it up to the calling user when and where (i.e. on which
+host) to generate the output fact, and makes no attempt at idempotence.
+If writing the ansible fact out to a file, the `to_json` filter must be
+used. By default, the output ansible fact is `ocp_pull_secrets`, but this
+can be customized if needed.
+
+Example playbooks are provided below which demonstrates the proper use of
+the `to_json` filter, various ways to load the access token, and ways to
+invoke this role in an idempotent way based on output facts or files.
+
+Requirements
+------------
+
+Ansible 2.7 or higher
+
+Red Hat Enterprise Linux 7 or equivalent
+
+Valid Red Hat Subscriptions
+
+Role Variables
+--------------
+
+### Required
+
+* `ocp_pull_secrets_offline_token` - Offline access token acquired from
+  https://cloud.redhat.com/openshift/token (login required), or an offline access token provided
+  by the authentication service providing the `ocp_pull_secrets_token_url` if the
+  default `sso.redhat.com` and `api.openshift.com` default services are not used.
+
+### Optional
+
+* `ocp_pull_secrets_out_var` - Name of the fact in which to store the gathered pull secrets JSON.
+  Default `ocp_pull_secrets`.
+* `ocp_pull_secrets_additional_auths` - Dictionary of additional authorizations to inject into the
+  gathered pull secrets, see usage example below. Default `{}` (empty dict).
+* `ocp_pull_secrets_no_log` - If true, prevent logging secrets to ansible output.
+  Default `true`.
+* `ocp_pull_secrets_token_url` - OpenID token endpoint used to convert the offline access token into
+  an authorization bearer token for use with the api endpoint URL.
+  Default `https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token`.
+* `ocp_pull_secrets_api_url` - API endpoint providing pull secrets for use by OCP.
+  Default `https://api.openshift.com/api/accounts_mgmt/v1/access_token`.
+
+Example Playbooks
+-----------------
+
+### Load access token from a file, write the pull secrets to a file
+
+The basic workflow of generating the pull secrets file, expected to be run on a
+single host, such as an OCP installation bootstrap/bastion host.
+
+```yaml
+- name: Load offline access token from file and write pull secrets to another file
+  hosts: ocp_pull_secrets_host
+  vars:
+    ocp_pull_secrets_offline_token: "{{ lookup('file', '/path/to/token.file') }}"
+  roles:
+    - role: oasis_roles.ocp_pull_secrets
+  tasks:
+    - name: Write pull secrets out to a secret place
+      copy:
+        # the to_json filter is required
+        # if outputting for use in the OCP installer, do not indent the output
+        # json, use the to_nice_json filter, or otherwise reformat it
+        content: "{{ ocp_pull_secrets | to_json }}"
+        dest: "/path/to/pull_secrets.json"
+        # "become" and/or set file modes appropriately to keep the secrets safe
```

Comment:
We could just go ahead and put a `mode: 0600` in this example since it will likely be copied as-is and it's easy to forget to read the comments.

Reply:
That makes sense, but this is intentionally vague since there are countless ways to output this fact. The main point is not how to safely write a file, but that you need to use `| to_json` and not `| to_nice_json` or whatever else.

I was hesitant in this case to even use the copy module, in favor of some obvious dummy stand-in module, because I'm interested in being very clear that this module is in no way responsible for, and should not be used as guidance for, securing your data.